app/src/main/java/com/example/razli/weatherappsb/view/MainActivity.kt

```diff
@@ -63,15 +63,15 @@ class MainActivity : AppCompatActivity(), MainContract.View {
 
         adapter.setOnItemClickListener(object : MainAdapter.OnItemClickListener {
             override fun onItemClick(itemView: View, position: Int) {
-                val place = itemView.placeNameTextView.text.toString()
-                showAlertDialog(place)
+                openWeatherForecastActivity(itemView.placeNameTextView.text.toString())
```

Comment:
This is okay, since it is quite simple. Meaning Next Page is launched right away from the View (without the need for Presenter's help).

But in the future, if there is another step, like Tracking Events, you should let the Presenter do the job of launching the View and also do the Tracking.